src/SDKs/Azure.Core/data-plane/Azure.Core/ResponseOfT.cs

```diff
@@ -8,34 +8,21 @@ namespace Azure
 {
     public readonly struct Response<T> : IDisposable
     {
-        private readonly Response _response;
-
         public Response(Response response, T parsed)
         {
-            _response = response;
+            Raw = response;
```

Comment:
I like the idea, but I'm having a hard time getting past the name.   `Raw` to me feels like `the source of something that has been transformed/changed/formatted.`   This really feels like `Response<T>` is really a composite that contains an `HttpResponse` and some kind of value.    

I might suggest renaming this class to `Result<T>` or `QueryResult<T>`  or something less clunky and then having  `Response` and `Value` properties.    In my brain, a `Response` is the thing that an HTTP request gave us.   Once we start doing things to it, I would view it as a result of some kind,  where the response is sent to some operation to help evaluate it and the result of that is a `Value` in this case.

If we're set on this name for it, then maybe we should consider changing `Raw` to something like `HttpResponse` or similar.    I know that particular name is easy to conflate with the built-in `HttpResponse` type, but it does have lot more semantic meaning in the context than `Raw`.

Reply:
I agree Raw is not a great name, but QueryResult\<T\> is worse :-)

Reply:
Maybe instread of Raw we could call it "Details"?

Reply:
I'd prefer an implicit cast to Response over Raw

Reply:
Of all options I still prefer `Raw` while agreeing it's not the best.


> I'd prefer an implicit cast to Response over Raw

It makes some code way more verbose:
``` C#
// Raw
response.Raw.Headers.TryGetValue("x-ms-client-request-id", out string requestId);
// Cast
var httpResponse = (Response)response;
response.Headers.TryGetValue("x-ms-client-request-id", out string requestId);
```